crates/biome_js_analyze/src/assist/source/use_sorted_interface_members.rs

```diff
@@ -0,0 +1,200 @@
+use biome_string_case::comparable_token::ComparableToken;
+use std::borrow::Cow;
+
+trait TsTypeMemberName {
+    fn name(&self) -> SyntaxResult<Option<AnyJsObjectMemberName>>;
+}
+impl TsTypeMemberName for AnyTsTypeMember {
+    fn name(&self) -> SyntaxResult<Option<AnyJsObjectMemberName>> {
+        match self {
+            // Property signatures have names
+            Self::TsPropertySignatureTypeMember(prop) => prop.name().map(Some),
+            // Method signatures have names
+            Self::TsMethodSignatureTypeMember(method) => method.name().map(Some),
+            // Getter signatures have names
+            Self::TsGetterSignatureTypeMember(getter) => getter.name().map(Some),
+            // Setter signatures have names
+            Self::TsSetterSignatureTypeMember(setter) => setter.name().map(Some),
+            // Call signatures, construct signatures, and index signatures don't have sortable names
+            _ => Ok(None),
+        }
+    }
+}
+fn is_interface_members_sorted(
+    list: &TsTypeMemberList,
+    comparator: impl Fn(&ComparableToken, &ComparableToken) -> std::cmp::Ordering,
+) -> bool {
+    // Collect all sortable members with their keys
+    let mut sortable_keys: Vec<ComparableToken> = Vec::new();
+
+    for member in list.iter() {
+        if let Ok(Some(name)) = member.name()
+            && let Some(token_text) = name.name()
+        {
+            sortable_keys.push(ComparableToken::new(token_text));
+        }
+    }
+
+    // Check if the sortable keys are already sorted
+    for i in 1..sortable_keys.len() {
+        if comparator(&sortable_keys[i - 1], &sortable_keys[i]) == std::cmp::Ordering::Greater {
+            return false;
+        }
+    }
+
+    true
+}
+fn sort_interface_members(
+    list: &TsTypeMemberList,
+    comparator: impl Fn(&ComparableToken, &ComparableToken) -> std::cmp::Ordering,
+) -> TsTypeMemberList {
+    let mut sortable_members = Vec::new();
+    let mut non_sortable_members = Vec::new();
+
+    // Separate sortable from non-sortable members
+    for member in list.iter() {
+        if let Ok(Some(name)) = member.name() {
+            if let Some(token_text) = name.name() {
+                sortable_members.push((member, ComparableToken::new(token_text)));
+            } else {
+                // Name exists but is not sortable (computed/dynamic)
+                non_sortable_members.push(member);
+            }
+        } else {
+            // No name (call signatures, construct signatures, index signatures)
+            non_sortable_members.push(member);
+        }
+    }
+
+    // Sort the sortable members
+    sortable_members.sort_by(|(_, a), (_, b)| comparator(a, b));
+
+    // Combine: all sortable members first, then all non-sortable members
+    let mut new_members: Vec<AnyTsTypeMember> = sortable_members
+        .into_iter()
+        .map(|(member, _)| member)
+        .collect();
+
+    new_members.extend(non_sortable_members);
```

Comment:
I think there's a bug here. We claim that some members aren't sortable, however we are moving them to the last place, even when they aren't.